pkg/registry/apis/provisioning/resources/folderid.go

```diff
@@ -0,0 +1,76 @@
+package resources
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"path"
+	"regexp"
+	"strings"
+)
+
+var disallowedKubernetesCharacters = regexp.MustCompile(`[^-a-z0-9]`)
+
+// FolderID contains the identifier data for a folder.
+type FolderID struct {
+	// Title is the human-readable name created by a human who wrote it.
+	Title string
+	// KubernetesName represents the name the folder should have, derived from the title.
+	// It contains a suffix calculated from the path of the folder.
+	KubernetesName string
+	// Path is the full path to the folder, as given to the parse function.
+	Path string
+}
+
+func ParseFolderID(dirPath string) FolderID {
+	clean := strings.Trim(path.Clean(dirPath), "/")
+	return FolderID{
+		Title:          calculateFolderTitleFromPath(clean),
+		KubernetesName: calculateKubernetesNameFromPath(clean),
+		Path:           dirPath,
+	}
+}
+
+func calculateFolderTitleFromPath(cleanDirPath string) string {
+	titleIdx := strings.LastIndex(cleanDirPath, "/")
+	title := cleanDirPath
+	if titleIdx != -1 {
+		title = title[titleIdx+1:]
+	}
+	return title
+}
+
+func calculateKubernetesNameFromPath(cleanDirPath string) string {
```

Comment:
[style] I find the names excessively long.  I think we can spare the verb and go for shorter ones :) 

```suggestion
func idFromPath(cleanDirPath string) string {
```